codice/alliance
# Cal 67 - Chipping Service Assumes Base Image Is North Aligned

#### What does this PR do?
This is a re-work of the math behind the chipping service implementation.  The previous version worked on the assumption that the image is always north-aligned (NW and NE corners on the same parallel).  This version uses a linear transformation to solve that problem.

#### Who is reviewing it?
@bdeining 
@rzwiefel 
@jlcsmith 

#### How should this be tested?
1) build the bundle and install it into DDF
2) ingest a NITF
3) request a chip (<protocol>://<host>:<port>/services/catalog/sources/ddf.distribution/<resource-id>?qualifier=overview&transform=chip&location=POLYGON ((<lon1> <lat1>, <lon2> <lat2>, <lon3> <lat3>, <lon4> <lat4>, <lon1> <lat1>)))
4) verify that the returned image is the area requested in step 3.

#### Any background context you want to provide?
This bug was discovered as part of the development of CAL-53 which needs to be updated in a similar way.

#### What are the relevant tickets?
CAL-67, CAL-53

#### Screenshots (if appropriate)
#### Checklist:
